Snowflake.Data.Tests/SFConnectionPoolT.cs

```diff
@@ -504,6 +504,98 @@ public void TestConcurrentConnectionPoolingAsync()
             ConcurrentPoolingAsyncHelper(connStr, true);
         }
 
+        [Test]
+        public void TestRollbackTransactionOnPooledWhenExceptionOccurred()
+        {
+            SnowflakeDbConnectionPool.SetPooling(true);
+            SnowflakeDbConnectionPool.SetMaxPoolSize(10);
+
+            object firstOpenedSession = null;
+            using (var connection = new SnowflakeDbConnection())
+            {
+                connection.ConnectionString = ConnectionString;
+                connection.Open();
+                firstOpenedSession = connection.SfSession.sessionId;
+                connection.BeginTransaction();
+                Assert.Throws<SnowflakeDbException>(() =>
+                {
+                    using (var command = connection.CreateCommand())
+                    {
+                        command.CommandText = "invalid command will throw";
+                        command.ExecuteNonQuery();
+                    }
+                });
+            }
+
+            using (var connectionWithSessionReused = new SnowflakeDbConnection())
+            {
+                connectionWithSessionReused.ConnectionString = ConnectionString;
+                connectionWithSessionReused.Open();
+                Assert.AreEqual(firstOpenedSession, connectionWithSessionReused.SfSession.sessionId);
```

Comment:
Thanks for this assert, but still - it would be more maintainable if this pool is 1-sized.

Reply:
fixed according to your suggestion @sfc-gh-pfus, thx!